lib/common/cmdline.c

```diff
@@ -189,6 +189,16 @@ pcmk__cmdline_preproc(char **argv, const char *special) {
             /* Skip over leading dash */
             char *ch = argv[i]+1;
 
+            /* This looks like the start of a number, which means it is a negative
+             * number.  It's probably the argument to the preceeding option, but
+             * we can't know that here.  Copy it over and let whatever handles
+             * arguments next figure it out.
+             */
+            if (*ch != '\0' && *ch >= '1' && *ch <= '9') {
+                g_ptr_array_add(arr, g_strdup_printf("%s", argv[i]));
+                continue;
+            }
```

Comment:
I'm thinking we should check whether the entire option is numeric. Otherwise we lose e.g. "crm_mon -1i3" (though I really hope no one's doing that).

Reply:
Good idea.  Updated.